internal/lib/lib.go

```diff
@@ -176,9 +176,14 @@ func GetL7SharedPGName(vsName string) string {
 	return vsName
 }
 
-func GetL7PoolName(priorityLabel, namespace, ingName string, args ...string) string {
+func GetL7PoolName(priorityLabel, namespace, ingName, infrasetting string, args ...string) string {
 	priorityLabel = strings.ReplaceAll(priorityLabel, "/", "_")
-	poolName := NamePrefix + priorityLabel + "-" + namespace + "-" + ingName
+	var poolName string
+	if infrasetting != "" {
+		poolName = NamePrefix + infrasetting + "-" + priorityLabel + "-" + namespace + "-" + ingName
```

Comment:
K8s 1.19 onwards would we force people to use an Ingress class? If so - would they be forced to use an AviInfraSetting for the default ingress class or would it work by simply referring to the global settings (hence the else block) cc @monotosh-avi

Reply:
AviInfraSetting is not forced in any way, so we'll fall back to defaults.